base_domain_operator/models/base_domain_operator.py

```diff
@@ -30,12 +30,19 @@ def _operator_parent_of(self, leaf, expression):
 
         # this function is more of less a verbatim copy of
         # https://github.com/OCA/OCB/blob/10.0/odoo/osv/expression.py#L738
-        def parent_of_domain(left, ids, left_model, parent=None, prefix=''):
+        def parent_of_domain(left, right, left_model, parent=None, prefix=''):
+            if isinstance(right, basestring):
+                fun = left_model.name_search
```

Comment:
I still think you can do here `ids = [_id for _id, dummy in left_model.name_search(ids)]` and leave the rest of the file unchanged

Reply:
@hbrunn I am not sure that would be a good idea. While this will work for cases in which the `ids` is  a string (from the web client, but it will fail when the `ids` is actually an array of integers.

Reply:
that's why you do this only under the scope of this conditional